source/CIPlanetNode.cpp

```diff
@@ -22,6 +22,10 @@ void PlanetNode::draw(const std::shared_ptr<cugl::SpriteBatch>& batch,
 }
 
 void PlanetNode::update(float timestep, bool isLockingIn, int numLayers, bool canLockIn) {
+    if (_progressNodes.size() != _layers->size()) {
+        setLayers(_layers);
+    }
+    
```

Comment:
Why is this bit needed? The only time that the size of either of these is changed is in setLayers itself

Reply:
oh this is here so the progress circles show up as soon as the game starts. It's kind of weird because the progress nodes are children of game scene but are stored in planet node. So planet node needs to be added to the scene graph before the progress nodes can be added to the scene graph. its kind of hacky but i can explain a bit more if your interested